shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -877,13 +913,45 @@ SemanticsObject hitTest(float[] point) {
         boolean isFocusable() {
             int scrollableActions = Action.SCROLL_RIGHT.value | Action.SCROLL_LEFT.value
                     | Action.SCROLL_UP.value | Action.SCROLL_DOWN.value;
+            if (hasFlag(Flag.IS_EDGE) && value == null) {
+                return false;
+            }
             return (actions & ~scrollableActions) != 0
                 || flags != 0
                 || (label != null && !label.isEmpty())
                 || (value != null && !value.isEmpty())
                 || (hint != null && !hint.isEmpty());
         }
 
+        void collectEdges(List<SemanticsObject> edges) {
+            if (hasFlag(Flag.IS_EDGE)) {
+                edges.add(this);
+            }
+            if (children != null) {
+                for (int i = 0; i < children.size(); ++i) {
+                    children.get(i).collectEdges(edges);
+                }
+            }
+        }
+
+        String getRouteName() {
+            String name = null;
+            if (children != null) {
+                for (int i = children.size() - 1; i >= 0; --i) {
```

Comment:
Remind me why we want the last name and not the first one?

Reply:
We do want the first one if it is non null & non-empty, cleaned up and added comment